application/comit_node/src/http_api/rfc003/routes.rs

```diff
@@ -48,9 +51,10 @@ pub fn get_swap<T: MetadataStore<SwapId>, S: StateStore>(
         .map(|(swap_resource, actions)| {
             let mut response = HalResource::new(swap_resource);
             for action in actions {
-                let route = format!("{}/{}", swap_path(id), action);
-                response.with_link(action, route);
+                let link = new_hal_link(&id, action);
+                response.with_link(action.to_string(), link);
```

Comment:
:wrench: I think `with_link` takes anything that goes `Into<String>`. You could try omitting the `to_string()` here.